articles/why-node-js-is-popular/index.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: engineering-education
+status: publish
+published: true
+slug: why-node-js-is-popular
+title: Why is Node.Js wildly popular among developers?
+description: Node.js is defined as a JavaScript runtime built on Chrome's V8 JavaScript engine. Node.js can be simply referred to as a technology. It is mainly used for backend operations.
+author: geoffrey-mungai
+date: 2020-08-12T00:00:00-10:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/why-node-js-is-popular/hero.jpg
+    alt: computer image example nodejs
+---
+Node.js has been a household name among developers for a few years now. So, what is all this recent excitement around Node.js we've been all hearing lately? Why is it so popular? Let's look at all this and more.
+<!--more-->
+### What is Node.js?
+The definition of Node.js is quite complex. It's not a programming language nor is it a framework or a library. On their official [website](https://nodejs.org/en), Node.js is defined as a JavaScript runtime built on Chrome's V8 JavaScript engine. Node.js can be simply referred to as a technology. It is mainly used for backend operations and is/can be used together with frontend JavaScript frameworks like [Angular](https://angularjs.org/), [React](https://reactjs.org/), [Vue](https://vuejs.org/) and others. Node.js is similar in design to Ruby's [Event Machine](https://github.com/eventmachine/eventmachine) and Python's [Twisted](https://twistedmatrix.com/trac/).
+
+### How and why did Node.js gain popularity?
+Node.js was introduced into the developer world in 2009 by the initial writer Ryan Dahl. Back in 2009, Apache HTTP server was a popular web server. But, it had a major problem, it could not handle a lot of concurrent connections. When Node.js was created it was able to solve the Apache problem. It offered scalability and parallel code execution. It introduced a new logic and approach for JavaScript server-side development. This has helped many developers.
```

Comment:
`the Apache problem` ? Possibly a little vague. Reading the next line one can infer it to be scalability and parallel code execution but if we are using it as a Noun it implies implicit knowledge. `the problems Apache had` might be better. 

`It introduced a new logic and approach for JavaScript server-side development.` this is vague too, some words on what new logic/approach was introduced will be better.